components/camel-nats/src/test/java/org/apache/camel/component/nats/NatsProducerReplyToTimeoutTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.nats;
+
+import org.apache.camel.CamelExecutionException;
+import org.apache.camel.ExchangeTimedOutException;
+import org.apache.camel.builder.RouteBuilder;
+import org.junit.jupiter.api.Test;
+
+import static org.apache.camel.test.junit5.TestSupport.assertIsInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class NatsProducerReplyToTimeoutTest extends NatsTestSupport {
+
+    protected String startUri = "direct:start";
+    protected String middleUri = "nats:foo.middle?requestCleanupInterval=50";
```

Comment:
I think we should use `requestTimeout` that is what the other messaging component uses. And it has a default timeout of 20 seconds. So this means adding a new option to use that. As that option is some sort of internal job by nats to cleanup stale jobs.

Reply:
Added `requestTimeout` with default timeout of 20 seconds.